### ISSUE TITLE
GIoU Loss becomes NAN

### ISSUE THREAD

**Issue by godprincelp** [generalized-iou/g-tensorflow-models#3]
The GIoU loss becomes NaN when the gt boxes and predicted boxes are both decoded in center and size (xc, yc, w, h). 

I am wondering if the boxes have to decoded in two corners (x1, y1, x2, y2).

**Comment by weichen456** [comment]
I have encountered the same problem

**Comment by zhoujungis** [comment]
I write the code by myself and when the model( yolo3 ) was training several epoch, the giou loss became nan.

**Comment by shrezatofighi** [comment]
> The GIoU loss becomes NaN when the gt boxes and predicted boxes are both decoded in center and size (xc, yc, w, h).
> 
> I am wondering if the boxes have to decoded in two corners (x1, y1, x2, y2).

No they don't have to. NaN may occurs when the loss stability cannot be ensured, e.g. when the outputed values for w or h can be non-positive.

**Comment by sbastani** [comment]
The same problem here. Anyone has managed to solve it?

**Comment by notabee** [comment]
same problem
